### PR TITLE
Conductor post release comms

### PR DESCRIPTION
**Pull request by itsjustkevin**
Updates post release comms and refactors. 

Originally merged in https://github.com/flutter/flutter/pull/106120, reverted as changes caused an infra failure due to an improper merge.

### CODE REVIEW

File: dev/conductor/core/lib/src/globals.dart [hunk 1]
```diff
@@ -14,13 +14,36 @@ const String kForceFlag = 'force';
...
 
-const String kLuciPackagingConsoleLink = 'https://ci.chromium.org/p/flutter/g/packaging/console';
+const String kLuciPackagingConsoleLink =
+    'https://ci.chromium.org/p/flutter/g/packaging/console';
 
-const String kWebsiteReleasesUrl = 'https://docs.flutter.dev/development/tools/sdk/releases';
+const String kWebsiteReleasesUrl =
+    'https://docs.flutter.dev/development/tools/sdk/releases';
+
+const String discordReleaseChannel =
```
Comment (CaseyHillers) [inline dev/conductor/core/lib/src/globals.dart:31]: Global consts should be prefixed with k

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#begin-global-constant-names-with-prefix-k

File: dev/conductor/core/lib/src/state.dart [hunk 2]
```diff
@@ -77,7 +77,7 @@ String presentState(pb.ConductorState state) {
   buffer.writeln('\tCurrent git HEAD: ${state.engine.currentGitHead}');
   buffer.writeln('\tPath to checkout: ${state.engine.checkoutPath}');
   buffer.writeln(
-      '\tPost-submit LUCI dashboard: ${luciConsoleLink(state.releaseChannel, 'engine')}');
+      '\tPost-submit cocoon dashboard: ${globals.cocoonDashboard}');
```
Comment (CaseyHillers) [inline dev/conductor/core/lib/src/state.dart:80]: We're in a weird state. For the framework, you need to look at both LUCI and the Cocoon dashboard.

I expect the next beta cut we will be able to deprecate the MILO dashboards for release CI. We'll still need it for packaging.

File: dev/conductor/core/lib/src/globals.dart [hunk 3]
```diff
@@ -171,7 +196,8 @@ String getNewPrLink({
       repoLabel = 'Engine';
       break;
     default:
-      throw ConductorException('Expected repoName to be one of flutter or engine but got $repoName.');
+      throw ConductorException(
```
Comment (CaseyHillers) [inline dev/conductor/core/lib/src/globals.dart:199]: These line fixes seem fine to me, but I defer to @christopherfujino 

I prefer codebases with 120 line lengths, but I do see that the framework generally requires 80 (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters)

File: dev/conductor/core/lib/src/globals.dart (hunk 1 above)
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/globals.dart]: This directly conflicts with the Dart style guide in Effective Dart. 

https://dart.dev/guides/language/effective-dart/style#dont-use-prefix-letters

As this is a Dart project, I went with their style guide.

File: dev/conductor/core/lib/src/globals.dart (hunk 3 above)
Comment (christopherfujino) [reply dev/conductor/core/lib/src/globals.dart:184]: I defer to Casey :)

File: dev/conductor/core/lib/src/globals.dart [hunk 4]
```diff
@@ -14,13 +14,21 @@ const String kForceFlag = 'force';
 
 const List<String> kBaseReleaseChannels = <String>['stable', 'beta'];
 
-const List<String> kReleaseChannels = <String>[...kBaseReleaseChannels, FrameworkRepository.defaultBranch];
+const List<String> kReleaseChannels = <String>[
+  ...kBaseReleaseChannels,
+  FrameworkRepository.defaultBranch
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/globals.dart:19]:
```suggestion
  FrameworkRepository.defaultBranch,
```

File: dev/conductor/core/lib/src/state.dart (hunk 2 above)
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/state.dart:80]: This is the link for engine and framework post submits.  I typically utilize those dashboards for the post submit tests.  The packaging link has not changed and still utilizes the LUCI link. 

See next.dart line 340.

Comment (christopherfujino) [reply dev/conductor/core/lib/src/state.dart:80]: To clarify, this change is good, as cocoon should be the source of truth for engine post-submit

**Review by christopherfujino — APPROVED** [review]
code changes LGTM.

File: dev/conductor/core/lib/src/globals.dart (hunk 3 above)
Comment (itsjustkevin) [reply dev/conductor/core/lib/src/globals.dart:184]: This was due to VSCode linting.  Maybe I need to change my linter settings. Thanks for the heads up!

File: dev/conductor/core/lib/src/globals.dart (hunk 1 above)
Comment (CaseyHillers) [reply dev/conductor/core/lib/src/globals.dart]: Since this is in the Flutter repo, the Flutter repo style guide will supersede effective dart

Comment (itsjustkevin) [reply dev/conductor/core/lib/src/globals.dart]: That is a fair point, I'll make the change.

File: dev/conductor/core/lib/src/globals.dart (hunk 3 above)
Comment (christopherfujino) [reply dev/conductor/core/lib/src/globals.dart:184]: Yeah, by default the Flutter/dart plugin will call dart format on save, which is against flutter style's guide.

**Comment by itsjustkevin** [comment]
Updating to clear test errors in state.dart and version.dart.

**Pull request closed without merging**